lib/state.coffee

```diff
@@ -9,15 +10,19 @@ exports.findCommandBySignature = (signature) ->
 	return _.findWhere exports.commands, (command) ->
 		return command.signature.toString() is signature
 
-exports.getMatchCommand = (signature) ->
+exports.getMatchCommand = (signature, callback) ->
 
-	for command in exports.commands
+	# Omit wildcard command from the check
+	commands = _.reject exports.commands, (command) ->
+		return command.isWildcard()
 
-		# Omit wildcard command from the check
-		continue if command.isWildcard()
+	async.eachSeries commands, (command, done) ->
+		command.signature.matches signature, (result) ->
+			return done() if not result
+			return callback(null, command)
```

Comment:
I guess the only way to do it with the async library would be to run `done(result)` and then below
```coffee-script
(result) ->
	return callback(null, result) if result?
```

It's a bit of an abomination but this loop never throws an actual error